testing/integration-tests/src/full_client/blocks/mod.rs

```diff
@@ -235,14 +235,22 @@ async fn fetch_block_and_decode_extrinsic_details() {
     assert!(!timestamp.is_signed());
 
     // Next we expect our transfer:
-    let tx = block_extrinsics.get(1).unwrap();
-    tx.as_root_extrinsic::<node_runtime::Call>().unwrap();
-    let ext = tx
-        .as_extrinsic::<node_runtime::balances::calls::types::TransferAllowDeath>()
-        .unwrap()
-        .unwrap();
-    assert_eq!(ext.value, 10_000);
-    assert!(tx.is_signed());
+    let mut balance_transfer = None;
+
+    for tx in block_extrinsics {
```

Comment:
You may also be able to replace the loop with something like:

```rust
if let Some(ext) = block_extrinsics.find_first::<node_runtime::balances::calls::types::TransferAllowDeath>().unwrap() {

}
```

But it's good to aleo be testing `as_root_extrinsic` anyways :)

Reply:
yeah, it's already tested above I noticed now